hnn_core/network.py

```diff
@@ -348,8 +347,9 @@ def __init__(self, params, add_drives_from_params=False,
         # We want it to remain in each Network object, so that the user can
         # interrogate a built and simulated net. In addition, CellResponse is
         # attached to a Network during simulation---Network is the natural
-        # place to keep this information
-        self.gid_ranges = dict()
+        # place to keep this information. Order matters: cell gids first, then
+        # artifical drive cells
+        self.gid_ranges = OrderedDict()
```

Comment:
Just to be suuuper explicit that [`NetworkBuilder._gid_assign`](https://github.com/jonescompneurolab/hnn-core/blob/c91fdcc1444f5265373f0954a0db833921af50b4/hnn_core/network_builder.py#L371) assumes a specific order.